scripts/py_matter_idl/matter/idl/matter_idl_parser.py

```diff
@@ -3,23 +3,21 @@
 import dataclasses
 import functools
 import logging
+import sys
+from pathlib import Path
 from typing import Dict, Optional, Set
 
 from lark import Lark
 from lark.lexer import Token
 from lark.visitors import Transformer, v_args
 
 try:
-    from matter_idl.matter_idl_types import AccessPrivilege
+    from matter.idl.matter_idl_types import AccessPrivilege
 except ModuleNotFoundError:
-    import os
-    import sys
-    sys.path.append(os.path.dirname(
-        os.path.dirname(os.path.abspath(__file__))))
+    sys.path.append(str(Path(__file__).resolve().parent / ".." / ".."))
```

Comment:
Just curious why was this changed like this? Functionally it is the the same, but I just want to know why you converted to this method here vs just using another `os.path.dirname`. I am just trying to learn, I don't have any opinion one way or the other

Reply:
Just updating for new Python. In new Python code it's cleaner to use `pathlib` for path manipulation. Since I've been updating imports I've seen this "old" way of dealing with paths, so I've updated it. It has clear object oriented semantic.